File: app/src/main/java/co/tiagoaguiar/codelab/myapplication/MainActivity.java
package co.tiagoaguiar.codelab.myapplication;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

import co.tiagoaguiar.codelab.myapplication.adapter.MainAdapter;

public class MainActivity extends AppCompatActivity {

    private RecyclerView rvMain;

    @Override

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        rvMain = findViewById(R.id.main_rv);

        List<MainItem> mainItems = new ArrayList<>();
        mainItems.add(new MainItem(1, R.drawable.ic_imc, R.string.imc, Color.GREEN));
        mainItems.add(new MainItem(2, R.drawable.ic_baseline_add_reaction_24, R.string.tmb, Color.YELLOW));

        //1 -> Definir o comportamento de exibição do layout da recyclerview (mosaic, grid
        // ou linear(horizontal/vertical))
        rvMain.setLayoutManager(new GridLayoutManager(this, 2));
        MainAdapter adapter = new MainAdapter(mainItems);

        adapter.setListener(id -> {
            switch (id) {
                case 1:
                    startActivity(new Intent(MainActivity.this, ImcActivity.class));
                    break;
                case 2:
                    startActivity(new Intent(MainActivity.this, TmbActivity.class));
                    break;
            }
        });

        rvMain.setAdapter(adapter);
    }


}

File: app/src/main/java/co/tiagoaguiar/fitnesstracker/MainActivity.kt
package co.tiagoaguiar.fitnesstracker

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import co.tiagoaguiar.fitnesstracker.adapter.MainAdapter
import co.tiagoaguiar.fitnesstracker.adapter.OnItemClickListener
import co.tiagoaguiar.fitnesstracker.adapter.item.MainItem

class MainActivity : AppCompatActivity() {
    private lateinit var rvMain: RecyclerView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Thread.sleep(10000)
        installSplashScreen()
        setContentView(R.layout.activity_main)

        val mainItems = mutableListOf<MainItem>()
        mainItems.add(
            MainItem(
                id = 1,
                drawableId = R.mipmap.ic_woman_foreground,
                textStringId = R.string.label_imc,
                color = Color.GREEN
            )
        )
        mainItems.add(
            MainItem(
                id = 2,
                drawableId = R.mipmap.ic_working_out_foreground,
                textStringId = R.string.label_tmb,
                color = Color.GREEN
            )
        )

        val adapter = MainAdapter(mainItems) { id ->
            when (id) {
                1 -> startActivity(Intent(this@MainActivity, ImcActivity::class.java))
                2 -> startActivity(Intent(this@MainActivity, TmbActivity::class.java))
            }
            Log.i("Teste", "clicou $id!")
        }
        rvMain = findViewById(R.id.rv_main)
        rvMain.adapter = adapter
        //1 -> Definir o comportamento de exibição do layout da recyclerview (mosaic, grid
        // ou linear(horizontal/vertical))
        rvMain.layoutManager = GridLayoutManager(this, 2)


    }

}